src/sigutils/lib.c

```diff
@@ -126,6 +128,30 @@ su_lib_fftw_strategy(void)
   return su_measure_ffts ? FFTW_MEASURE : FFTW_ESTIMATE;
 }
 
+SU_FFTW(_plan)
+su_lib_plan_dft_1d(int n, SU_FFTW(_complex) *in, SU_FFTW(_complex) *out,
+        int sign, unsigned flags)
+{
+  SU_FFTW(_plan) plan;
+  int nthreads;
+
+  if (n < 32768)
+    nthreads = 1;
+  else if (n == 32768)
+    nthreads = 2;
+  else
+    nthreads = 4;
```

Comment:
I am going to mere this as-is. But perhaps, as an improvement for the future, we can choose the number of threads depending on the number of available cores in a system.